File: LeetCode/Easy/0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        dp = [cost[0], cost[1]]

        for i in range(2, len(cost)):
            dp.append(min(dp[-1], dp[-2]) + cost[i])

        return min(dp[-1], dp[-2])

File: LeetCode/Easy/0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.rs
impl Solution {
    pub fn min_cost_climbing_stairs(cost: Vec<i32>) -> i32 {
        let mut dp = Vec::from([cost[0], cost[1]]);
        for i in 2..cost.len() {
            dp.push(cost[i] + dp[i - 1].min(dp[i - 2]))
        }
        dp[dp.len() - 1].min(dp[dp.len() - 2])
    }
}
